Automattic/jetpack
# Subscription block : update email control  type

## Changes proposed in this Pull Request:

Here we're updating the type of the email input control to "email" to:

1. bring it in line with similar controls in this method
2. ensure that WPCOM sites, which render this input, look dandy 🎉 

Since the [block's CSS styles](https://github.com/Automattic/jetpack/blob/d001afc08e7a70209cfdaafbbca14a91b0e65a25/projects/plugins/jetpack/extensions/blocks/subscriptions/view.scss#L28) specifically target `type="email"` however, it's not getting the `width: 100%` for which it is destined!

### Before
<img width="692" alt="Screen Shot 2021-04-27 at 9 34 03 pm" src="https://user-images.githubusercontent.com/6458278/116235320-09262480-a7a1-11eb-88de-744eb1b1b7fa.png">

### After
<img width="705" alt="Screen Shot 2021-04-27 at 9 33 38 pm" src="https://user-images.githubusercontent.com/6458278/116235307-06c3ca80-a7a1-11eb-8aa7-468c2faadd3c.png">
Fixes #19563

## Does this pull request change what data or activity we track or use?
No.

## Testing instructions

1. Apply this patch to WPCOM
2. Create a new post and add a Subscription block. 
3. The email input control should have a width of `100%` on both the editor and the frontend.
4. Repeat for a non-WPCOM site to make sure we haven't broken anything.
5. Always salt your water before boiling pasta. For flavour.
